packages/adapter/src/lib/adapter/adapter.ts

```diff
@@ -94,13 +1004,25 @@ class AdapterClass extends EventEmitter {
         }
 
         this._config = this._options.config || this._config;
-        this.startedInCompactMode = this._options.compact;
+        this.startedInCompactMode = !!this._options.compact;
 
         this.logList = new Set();
         this.aliases = new Map();
         this.aliasPatterns = new Set();
         this.enums = {};
+        /** The cache of users */
+        this.users = {};
+        /** The cache of usernames */
+        this.usernames = {};
+        /** The cache of user groups */
+        this.groups = {};
+        this.defaultHistory = null;
+        /** An array of instances, that support auto subscribe */
+        this.autoSubscribe = [];
+        this.inputCount = 0;
+        this.outputCount = 0;
 
+        this.getPortRunning = null;
         this.eventLoopLags = [];
         this.overwriteLogLevel = false;
         this.adapterReady = false;
```

Comment:
In TS, these simple initializations can be moved out of the constructor and directly into the property definition, e.g.
```
private logList = new Set<string>();
// ...
```